app/cloud.go

```diff
@@ -156,3 +157,30 @@ func (a *App) SendCloudTrialEndedEmail() *model.AppError {
 
 	return nil
 }
+
+// GetCloudUsageForMessages returns posts usage percentage against total available limit on Cloud.
+// Usage percentage is returned in multiples of 10 eg. 0, 10, 20
+func (a *App) GetCloudUsageForMessages(userID string) (int, *model.AppError) {
+	// Fetch limit
+	limits, err := a.Cloud().GetCloudLimits(userID)
+	if err != nil {
+		return 0, model.NewAppError("GetCloudUsageForMessages", "app.channel.GetCloudLimits.internal_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Fetch messages count
+	c, err := a.Srv().Store.Post().AnalyticsPostCount(&model.PostCountOptions{ExcludeDeleted: true})
+	if err != nil {
+		return 0, model.NewAppError("GetCloudUsageForMessages", "app.channel.AnalyticsPostCount.internal_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	// Return usage %
+	max := float64(*limits.Messages.History)
```

Comment:
@neallred What's the difference between a limit of `0` and a limit of `nil`?

Reply:
There shouldn't be a limit of 0 for messages, but if there were, it would mean they're not allowed to post any messages. I think "make 0 meaningful" by having `0` mean no limit could be useful for message history, but I could see 0 being a valid limit for other resources now or in the future, so I didn't want to have an inconsistency where for some limits 0 means none allowed but for others it means no limit.

Reply:
Thanks for clarifying! SGTM!